### PR TITLE
[DPE-4670] Reorder dashboards per relevance + integration tests clean-up

### PR DESCRIPTION
**Pull request by phvalguima**
Cleans up left-overs and do some experiments with our integration tests.

Changes:
1) Reorder dashboard: moves `Threadpool` dashboard all the way to the bottom on Grafana
2) Remove `upgrade_to_local` from main `ha.helpers.py`, as it is only used by upgrades
3) Downgrade all small upgrade tests to use GH runners
4) Move `test_storage.py` to bigger runners and see if we can stabilize these tests

New Threadpool dashboard section looks similar to a visual representation of `/_threadpool` API:

Screenshot from 2024-06-24 23-08-31

### CODE REVIEW

File: lib/charms/opensearch/v0/opensearch_base_charm.py [hunk 1]
```diff
@@ -1553,9 +1552,9 @@ def _get_prometheus_labels(self) -> Optional[Dict[str, str]]:
         try:
             if not self.opensearch.roles:
                 return None
+            taggable_roles = ClusterTopology.generated_roles() + ["voting"]
```
Comment (phvalguima) [inline lib/charms/opensearch/v0/opensearch_base_charm.py:1555]: As requested on PR #328, by @Mehdi-Bendriss

File: lib/charms/opensearch/v0/opensearch_base_charm.py [hunk 2]
```diff
@@ -1553,9 +1552,9 @@ def _get_prometheus_labels(self) -> Optional[Dict[str, str]]:
...
             roles = set(
-                role if role in COS_TAGGABLE_ROLES else "unrecognized"
-                for role in self.opensearch.roles
+                role if role in taggable_roles else "other" for role in self.opensearch.roles
```
Comment (phvalguima) [inline lib/charms/opensearch/v0/opensearch_base_charm.py:1557]: Renamed to `other`, as requested by @Mehdi-Bendriss  on #328

File: tests/integration/ha/helpers.py [hunk 3]
```diff
@@ -595,65 +593,3 @@ async def assert_restore_indices_and_compare_consistency(
     # We expect that new_count has a loss of documents and the numbers are different.
     # Check if we have data but not all of it.
     assert 0 < new_count < original_count
-
-
-async def assert_upgrade_to_local(
```
Comment (phvalguima) [inline tests/integration/ha/helpers.py:600]: Also present on `ha/upgrades/helpers.py`

**Review by reneradoi — APPROVED** [review]

File: tests/integration/upgrades/test_manual_large_deployment_upgrades.py [hunk 4]
```diff
@@ -34,7 +34,7 @@
 
 
 @pytest.mark.skip(reason="Fix with DPE-4528")
-@pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "xlarge"])
+# @pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "xlarge"])
```
[the remark this answers is not in the document]
Comment (phvalguima) [reply tests/integration/upgrades/test_manual_large_deployment_upgrades.py:37]: @Mehdi-Bendriss this test will be updated once we finish DPE-4528. I will try removing the self-hosted runner then.

**Pull request merged**